Jetsnack/app/src/main/java/com/example/jetsnack/ui/snackdetail/SnackDetail.kt

```diff
@@ -112,11 +121,36 @@ fun SnackDetail(
 
 @Composable
 private fun Header() {
+    val brushColors = JetsnackTheme.colors.tornado1
+
+    val infiniteTransition = rememberInfiniteTransition(label = "background")
+    val targetOffset = with(LocalDensity.current) {
+        1000.dp.toPx()
+    }
+    val offset by infiniteTransition.animateFloat(
+        initialValue = 0f,
+        targetValue = targetOffset,
+        animationSpec = infiniteRepeatable(
+            tween(50000, easing = LinearEasing),
+            repeatMode = RepeatMode.Reverse
+        ),
+        label = "offset"
+    )
+
+    val brushSize = 400f
+    val brush = Brush.linearGradient(
+        colors = brushColors,
+        start = Offset(offset, offset),
+        end = Offset(offset + brushSize, offset + brushSize),
+        tileMode = TileMode.Mirror
+    )
     Spacer(
         modifier = Modifier
             .height(280.dp)
             .fillMaxWidth()
-            .background(Brush.horizontalGradient(JetsnackTheme.colors.tornado1))
+            .blur(40.dp)
+            .background(brush)
```

Comment:
Do you think we could change this to custom drawing that would prevent recomposing this header and would just redraw?
<img width="413" alt="image" src="https://github.com/user-attachments/assets/8fe1fd93-5cf7-423a-bd70-3ce06b8e07a5">


```
            .drawWithCache {
                val brushSize = 400f
                val brush = Brush.linearGradient(
                    colors = brushColors,
                    start = Offset(offset, offset),
                    end = Offset(offset + brushSize, offset + brushSize),
                    tileMode = TileMode.Mirror
                )
                onDrawBehind {
                    drawRect(brush)
                }
            }
```

Reply:
Good idea, thanks Tomas.